fw/http_parser.c

```diff
@@ -2265,45 +2359,59 @@ __req_parse_cache_control(TfwHttpReq *req, unsigned char *data, size_t len)
 	}
 
 	__FSM_STATE(Req_I_CC_n) {
+		if (unlikely(!parser->cache_control.dir_allowed))
+			__FSM_EXIT(TFW_BLOCK);
 		TRY_STR_LAMBDA("no-cache", {
-			parser->_acc = TFW_HTTP_CC_NO_CACHE;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_NO_CACHE;
 		}, Req_I_CC_n, Req_I_CC_Flag);
 		TRY_STR_LAMBDA("no-store", {
-			parser->_acc = TFW_HTTP_CC_NO_STORE;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_NO_STORE;
 		}, Req_I_CC_n, Req_I_CC_Flag);
 		TRY_STR_LAMBDA("no-transform", {
-			parser->_acc = TFW_HTTP_CC_NO_TRANSFORM;
+			parser->cache_control.dir_flag = TFW_HTTP_CC_NO_TRANSFORM;
 		}, Req_I_CC_n, Req_I_CC_Flag);
 		TRY_STR_INIT();
 		__FSM_I_JMP(Req_I_CC_Ext);
 	}
 
 	__FSM_STATE(Req_I_CC_o) {
+		if (unlikely(!parser->cache_control.dir_allowed))
+			__FSM_EXIT(TFW_BLOCK);
```

Comment:
Hello! I wonder a little that the conditions `!parser->cache_control.dir_allowed` is tested at several states.
It is set to `true` at `Req_I_CC_start`. It can become `false` during parsing? Or the parser can enter these states avoiding `Req_I_CC_start` and with `false` value for condition? Believing to the given name `dir_allowed`, the check probably better be outside this function at all?